builtInFunctions/esdtGlobalSettings.go

```diff
@@ -88,9 +88,19 @@ func (e *esdtGlobalSettings) ProcessBuiltinFunction(
 		return nil, ErrOnlySystemAccountAccepted
 	}
 
-	esdtTokenKey := append(e.keyPrefix, vmInput.Arguments[0]...)
+	var err error
+	var systemSCAccount vmcommon.UserAccountHandler
+	if !bytes.Equal(core.SystemAccountAddress, vmInput.RecipientAddr) || check.IfNil(dstAccount) {
```

Comment:
Can we extract this if/else inside a func?
Also, can we have a unit test for this?

Reply:
done